### PR TITLE
Issue #464: Replaces Grunt with NPM scripts

### LINKED ISSUE #464: get rid of grunt
Replace it with npm tasks and then make the custom tasks in the /task folder to be just node tasks

### PR DESCRIPTION
**Pull request by jakeclements**
Fixes #464 
- Replaces Grunt with NPM scripts
- Removes text-limit on changelog messages
- Updates contributing and readme docs with run-script commands

### CODE REVIEW

**Comment by matteofigus** [comment]
wow @jakeclements this is an amazing work!
image

Give us some time to review but I just wanted to start saying thank you for this!

**Comment by jakeclements** [comment]
Thanks for the review @matteofigus, say the word once your PRs are in and I'll merge this.

**Comment by nickbalestra** [comment]
Wohoo, super excited for this. 
Thanks 💯  @jakeclements for the 💅 😍 , awesome stuff!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -24,11 +24,11 @@ You need to be enabled for doing this.
 * `master` should be all green. If not, make it green first.
 * git checkout master
 * git pull --tags
-* Run `grunt version:<versionType>` for new version.
+* Run `npm run version -- --type="<versionType>"` for new version.
```
Comment (matteofigus) [inline CONTRIBUTING.md]: I was thinking that for the sake of simplicity, we could simplify this by having `npm run patch`, `npm run minor` and `npm run major`? What do you think

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,11339 @@
+{
```
Comment (matteofigus) [inline package-lock.json]: We gitignored the package-lock in master, can you sync with that so we can remove this?

File: CONTRIBUTING.md (hunk 1 above)
Comment (jakeclements) [reply CONTRIBUTING.md]: I agree, the current script is unnecessarily verbose.
I'm happy to update this so we can keep it within a single PR.

**Comment by jakeclements** [comment]
@matteofigus thanks for the review, no problem about the timeframe. 
I've updated based on the requested changes, take a look when you get a chance and let me know if you would like me to merge.

**Comment by matteofigus** [comment]
Welcome to the contributors! 👍

**Pull request merged**